frontend/packages/core-auth/src/components/AuthWrapper.tsx

```diff
@@ -142,8 +156,20 @@ const AuthWrapper: React.FC<AuthWrapperProps> = ({
       if (refreshTokenInterval.current)
         clearInterval(refreshTokenInterval.current);
     };
-  }, [tokenResponse?.refreshToken, tokenResponse?.expiresIn]);
+  }, [tokenResponse?.refreshToken, tokenResponse?.expiresIn, discovery]);
```

Comment:
`JSON.stringify(discovery)`

Reply:
what's the pattern behind this? anytime it's an object? even when no single properties are being addressed?

Reply:
@neb42

Reply:
When useEffect checks is dependencies have changed it does a shallow comparison. So doing a shallow comparison of an instance of `TokenResponse` will always be different on each render. Because of this you want to always compare something that will do a shallow comparison properly (eg. string, number, boolean, null, undefined).

When deciding between `tokenResponse?.refreshToken` or `JSON.stringify(tokenResponse)` it's really about reasoning why you want the useEffect block to run. When you write the token response to local storage you want to write the entire object so you want to run it if any of `JSON.stringify(tokenResponse)` changes. In this block we don't want to reset the refresh interval if the access token changes, because it doesn't use the access token. It's using the refresh token, so `tokenResponse?.refreshToken` makes sense.

For `discovery` here we're using the full discovery object so we want to stringify that.

Reply:
But discovery doesn't change right? it's either undefined or in it's final state, so we're testing for pure existance in that case. wouldn't it then be okay to not use JSON.stringify?

Reply:
If we want to test for the existence of `discovery`, we should do something like `Boolean(discovery)`. Using just `discovery`, I'd expect this to run on every rerender because it would check if `discoveryNew === discoveryOld` which should return false every time.

I'd say we do want to check the full discovery object because of this line:
```
 const resp = await tokenResponse.refreshAsync(refreshConfig, discovery);
```
If discovery changes (even though it shouldn't) we want to rerun this useEffect so that this line uses the new discovery.